test/torchaudio_unittest/models/ctc_decoder/ctc_decoder_test.py

```diff
@@ -72,14 +80,53 @@ def test_timesteps_shape(self, use_lexicon):
             self.assertEqual(result.tokens.shape, result.timesteps.shape)
 
     def test_no_lm_decoder(self):
-        """Check that using no LM produces the same result as using an LM with 0 lm_weight"""
-        kenlm_decoder = self._get_decoder(lm_weight=0)
-        zerolm_decoder = self._get_decoder(use_lm=False)
+        """Check that the following produce the same result
+        - using no LM (C++ based implementation)
+        - using no LM (Custom Python based wrapper)
+        - using a (Ken)LM with 0 weight
+        """
+        from .ctc_decoder_utils import CustomZeroLM
 
         emissions = self._get_emissions()
+        custom_zerolm = CustomZeroLM()
+        zerolm_decoder_custom = self._get_decoder(lm=custom_zerolm)
+        zerolm_decoder_cpp = self._get_decoder(lm=None)
+        kenlm_file = get_asset_path("decoder/kenlm.arpa")
+        kenlm_decoder = self._get_decoder(lm=kenlm_file, lm_weight=0)
+
+        zerolm_custom_results = zerolm_decoder_custom(emissions)
+        zerolm_cpp_results = zerolm_decoder_cpp(emissions)
         kenlm_results = kenlm_decoder(emissions)
-        zerolm_results = zerolm_decoder(emissions)
-        self.assertEqual(kenlm_results, zerolm_results)
+
+        self.assertEqual(zerolm_cpp_results, zerolm_custom_results)
+        self.assertEqual(zerolm_cpp_results, kenlm_results)
+
+    def test_custom_kenlm_decoder(self):
+        """Check that creating a custom Python KenLM wrapper produces same results as C++ based KenLM"""
+        emissions = self._get_emissions()
+
+        kenlm_file = get_asset_path("decoder/kenlm.arpa")
+        custom_kenlm = self._get_custom_kenlm(kenlm_file)
+        kenlm_decoder_custom = self._get_decoder(lm=custom_kenlm)
+        kenlm_decoder_cpp = self._get_decoder(lm=kenlm_file)
+
+        kenlm_custom_results = kenlm_decoder_custom(emissions)
+        kenlm_cpp_results = kenlm_decoder_cpp(emissions)
+
+        self.assertEqual(kenlm_custom_results, kenlm_cpp_results)
+
+    def test_custom_nnlm_decoder(self):
+        """Check that "foo" biased NNLM only produces "foo" words"""
+        emissions = self._get_emissions()
+        lm_dict = get_asset_path("decoder/nnlm_dict.txt")
+
+        custom_nnlm = self._get_custom_foolm(lm_dict)
+        nnlm_decoder = self._get_decoder(lm=custom_nnlm, lm_dict=lm_dict)
+        nnlm_results = nnlm_decoder(emissions)
+
+        for result in nnlm_results:
+            for word in result[0].words:
+                assert word == "foo"
```

Comment:
Can we unnest these two for-loops?
If either of `nnlm_resutls`  or `.words` are empty, then no assertion is performed and the tests pass.

something like
```
expected = ["foo" for _ in range(N) for _ in range(M)]  # N and M corresponds to expected length, but hard coded
output = [res[0].words for res in nnlm_results]
assert expected == output
```